src/microbe_stage/systems/SiderophoreSystem.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        this.worldSimulation = worldSimulation;
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        if (!entity.Has<SiderophoreProjectile>())
+            return;
+
+        ref var projectile = ref entity.Get<SiderophoreProjectile>();
+
+        if (projectile.IsUsed)
+            return;
+
+        ref var collisions = ref entity.Get<CollisionManagement>();
+
+        if (!projectile.ProjectileInitialized)
+        {
+            projectile.ProjectileInitialized = true;
+
+            collisions.StartCollisionRecording(Constants.MAX_SIMULTANEOUS_COLLISIONS_TINY);
+        }
+
+        // Check for active collisions that count as a hit and use up this projectile
+        var count = collisions.GetActiveCollisions(out var activeCollisions);
+        for (int i = 0; i < count; ++i)
+        {
+            ref var collision = ref activeCollisions![i];
+
+            if (!HandleSiderophoreCollision(ref collision, in worldSimulation, entity
+                    .Get<SiderophoreProjectile>().Sender))
+            {
+                continue;
+            }
+
+            // Applied a damaging hit, destroy this toxin
+            // TODO: We should probably get some *POP* effect here.
+
+            // Expire right now
+            ref var timedLife = ref entity.Get<TimedLife>();
+            timedLife.TimeToLiveRemaining = -1;
+
+            ref var physics = ref entity.Get<Physics>();
+
+            // TODO: should this instead of disabling the further collisions be removed from the world immediately
+            // to cause less of a physics impact?
+            // physics.BodyDisabled = true;
+            physics.DisableCollisionState = Physics.CollisionState.DisableCollisions;
+
+            // And make sure the flag we check for is set immediately to not process this projectile again
+            // (this is just extra safety against the time over callback configuration not working correctly)
+            projectile.IsUsed = true;
+
+            // Only deal damage at most to a single thing
+            break;
+        }
+    }
+
+    private static bool HandleSiderophoreCollision(ref PhysicsCollision collision,
+        in IWorldSimulation worldSimulation, Entity sender)
+    {
+        var target = collision.SecondEntity;
+
+        // Skip if hit something that's not a chunk
+        if (!target.Has<ChunkConfiguration>())
+            return false;
+
+        ref var configuration = ref target.Get<ChunkConfiguration>();
+
+        var iron = SimulationParameters.Instance.GetCompound("iron");
+
+        if (configuration.Compounds == null)
+            return false;
+
+        // Check if it is the big iron chunk
+        if (configuration.Name == "BIG_IRON_CHUNK")
+        {
+            if (configuration.Compounds[iron].Amount > 0)
+            {
+                var smallIronChunk = SimulationParameters.Instance.GetBiome("default")
+                    .Conditions.Chunks["ironSmallChunk"];
```

Comment:
This should be fetched in this class constructor.

Reply:
Like this?
![image](https://github.com/user-attachments/assets/aa6a4083-9358-4c4b-a412-7e438190a736)
I get https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0120?f1url=%3FappId%3Droslyn%26k%3Dk(CS0120) when trying to access it later.

Reply:
Did you make the method non-static? I think for another thing I mentioned you'll need to make the method not static to have access to the class fields.